crypto/ocsp/ocsp_test.cc

```diff
@@ -1088,3 +1599,116 @@ TEST(OCSPTest, TestNotValidResponseOCSP) {
                             &basic_response,
                             &server_cert_chain);
 }
+
+
+// Test valid OCSP response hashed with sha256
+TEST(OCSPTest, TestGoodOCSP_SHA256) {
+  bssl::UniquePtr<OCSP_BASICRESP> basic_response;
+  bssl::UniquePtr<STACK_OF(X509)> server_cert_chain;
+  ExtractAndVerifyBasicOCSP(ocsp_response_sha256_der,
+                            OCSP_RESPONSE_STATUS_SUCCESSFUL,
+                            ca_cert,
+                            server_cert,
+                            OCSP_VERIFYSTATUS_SUCCESS,
+                            &basic_response,
+                            &server_cert_chain);
+
+  int status = 0;
+  ASN1_GENERALIZEDTIME *thisupd, *nextupd;
+  CheckOCSP_CERTSTATUS(&basic_response,
+                       &server_cert_chain,
+                       EVP_sha256(),
+                       1,
+                       &status,
+                       &thisupd,
+                       &nextupd);
+  ASSERT_EQ(V_OCSP_CERTSTATUS_GOOD, status);
+
+  // If OCSP response is verifiable and all good, an OCSP client should check
+  // time fields to see if the response is still valid
+
+  // Check before OCSP was last updated connection timestamp
+  time_t connection_time = invalid_before_ocsp_update_time_sha256;
+  ASSERT_EQ(1, X509_cmp_time(thisupd, &connection_time));
+  ASSERT_EQ(1, X509_cmp_time(nextupd, &connection_time));
```

Comment:
Will any source code of awslc help validate these time fields? If so, clock skew (e.g. [JDK bug](https://bugs.java.com/bugdatabase/view_bug.do?bug_id=6744888)) can be an issue?

Reply:
The best time comparison function I could find in awslc was `X509_cmp_time`, which unforunately [only checks if the value is larger or smaller](https://github.com/awslabs/aws-lc/blob/main/crypto/x509/x509_vfy.c#L1969-L1973).. That's why I set up a second before and after the time fields to verify that the ASN1 parsing of time fields is correct.
I'm not quite sure if clock skew will be an issue, but I would imagine our main job is to verify that the correct time fields are parsed and returned when reading a OCSP response. From my understanding, whether the time stamp that was parsed is within the correct time frame or not is up for the API user to decide (e.g. [s2n's OCSP usage](https://github.com/aws/s2n-tls/blob/main/tls/s2n_x509_validator.c#L483-L521)).